lib/src/Uncertainty/Distribution/SquaredNormal.cxx

```diff
@@ -0,0 +1,131 @@
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include <cstdlib>
+#include <cmath>
+
+#include "openturns/SquaredNormal.hxx"
+#include "openturns/Distribution.hxx"
+#include "openturns/ChiSquare.hxx"
+#include "openturns/SpecFunc.hxx"
+#include "openturns/Log.hxx"
+#include "openturns/OSS.hxx"
+#include "openturns/DistFunc.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/Matrix.hxx"
+#include "openturns/MatrixImplementation.hxx"
+#include "openturns/IdentityMatrix.hxx"
+//#include "openturns/NormalCopula.hxx"
+#include "openturns/ResourceMap.hxx"
+#include "openturns/RandomGenerator.hxx"
+#include "openturns/GaussKronrodRule.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(SquaredNormal)
+
+static const Factory<SquaredNormal> Factory_SquaredNormal;
+
+/* Constructor for multiD standard normal distribution */
+SquaredNormal::SquaredNormal()
+  : SquaredNormal(0.0, 1.0)
+{
+  setName("SquaredNormal");
+}
+
+/* Constructor */
+SquaredNormal::SquaredNormal(Scalar mu, Scalar sigma)
+  : ContinuousDistribution(), mu_(mu), sigma_(sigma), normal_(mu, sigma)
+{
+  setName("SquaredNormal");
+  computeRange();
+}
+
+/* Comparison operator */
+Bool SquaredNormal::operator ==(const SquaredNormal & other) const
+{
+  if (this == &other) return true;
+  return this->normal_ == other.normal_;
+}
+
+/* String converter */
+String SquaredNormal::__repr__() const
+{
+  OSS oss;
+  oss << "class=" << SquaredNormal::GetClassName()
+      << " squarenormal=" << this->normal_;
+  return oss;
+}
+
+String SquaredNormal::__str__(const String & offset) const
+{
+  OSS oss;
+  oss << getClassName() << " squared of " << this->normal_; 
+  return oss;
+}
+
+/* Virtual constructor */
+SquaredNormal * SquaredNormal::clone() const
+{
+  return new SquaredNormal(*this);
+}
+
+Point SquaredNormal::getRealization() const
+{
+  return Point(std::pow(this->mu_ + this->sigma_ * DistFunc::rNormal(), 2));
+}
+
+Sample SquaredNormal::getSample(const UnsignedInteger size) const
+{
+  Point xi = DistFunc::rNormal(size);
+  Sample result(size, 1);
+  for (UnsignedInteger i = 0; i < size; ++i)
+    result(i, 0) = std::pow((this->mu_ + this->sigma_ * xi[i]), 2); 
+  result.setName(getName());
+  result.setDescription(getDescription());
+  return result;
+}
+
+/* Get the CDF of the distribution */
+Scalar SquaredNormal::computeCDF(const Point & point) const
+{
+  Scalar x = point[0];
+  if (x <= 0.0) return 0.0;
+  return 0.5 * (SpecFunc::Erf((std::sqrt(x) + this->mu_) / (std::sqrt(2) * this->sigma_)) + SpecFunc::Erf((std::sqrt(x) - this->mu_) / (std::sqrt(2) * this->sigma_)));
```

Comment:
write std::sqrt(2.0) instead of sqrt(2) else it errors with vc10:
(SquaredNormal.cxx(111): error C2668: 'std::sqrt' : ambiguous call to overloaded function)

Reply:
I use M_SQRT2 now